solarforecastarbiter/io/reference_observations/common.py

```diff
@@ -209,16 +209,54 @@ def create_observation(api, site, variable, extra_params=None, **kwargs):
         return created
 
 
+def _utcnow():
+    return pd.Timestamp.now(tz='UTC')
+
+
+def get_last_site_timestamp(api, site_observations, end):
+    """Get the last value timestamp from the API for a site
+    as the minimum of the last timestamp for each observation at
+    that site. A limit of end - 7 days is set to avoid excessive
+    queries to external sources.
```

Comment:
When I first read this I got confused about if this function was actually querying the external sources. Maybe "A limit of end - 7 days is set to avoid excessive queries to external sources when the returned value is passed to fetch functions." I don't love that wording though. Ok to ignore too.